### PR TITLE
Keycloak panicking when attempt to encrypt JWT

### PR DESCRIPTION
**Pull request by SushmitaGoswami**
**Closes**: https://github.com/keycloak/keycloak/issues/32919
**Signed-off-by**: Sushmita Goswami [sushmita.g24@gmail.com](https://github.com/keycloak/keycloak/pull/sushmita.g24@gmail.com)

### CODE REVIEW

**Review by mposolda — COMMENTED** [review]
@SushmitaGoswami Thanks for the PR!

Added one minor note inline. Is it also possible to add automated test for this? Possibly candidate where it can be good to add the test could be the class `IdTokenEncryptionTest` .

@tnorimat Can I please ask you for the review as well since you are the author of the original commit https://github.com/keycloak/keycloak/commit/8225157a where the `if` condition was added. Do you see any issue with supporting the imported keys also for the encryption?

File: services/src/main/java/org/keycloak/keys/loader/ClientPublicKeyLoader.java [hunk 1]
```diff
@@ -76,17 +76,17 @@ public PublicKeysWrapper loadKeys() throws Exception {
         } else if (config.isUseJwksString()) {
             JSONWebKeySet jwks = JsonSerialization.readValue(config.getJwksString(), JSONWebKeySet.class);
             return JWKSUtils.getKeyWrappersForUse(jwks, keyUse);
-        } else if (keyUse == JWK.Use.SIG) {
+        } else if (keyUse == JWK.Use.SIG || keyUse == JWK.Use.ENCRYPTION) {
```
[the remark this answers is not in the document]
Comment (SushmitaGoswami) [reply services/src/main/java/org/keycloak/keys/loader/ClientPublicKeyLoader.java:79]: @mposolda Sure. I will update the PR and try adding a test. Just to be confirm, I see the flaky-tests which are reported are there for long, so I will not consider it for this PR. Please correct me if my assumption is wrong.

**Comment by tnorimat** [comment]
@mposolda Hello, I will reivew the PR as you requested.

**Review by tnorimat — APPROVED** [review]
LGTM if you change the point @mposolda suggested.

**Comment by SushmitaGoswami** [comment]
> LGTM if you change the point @mposolda suggested.

I will add the test soon and make the changes as mentioned.

**Comment by mposolda** [comment]
@SushmitaGoswami Cool, Thanks!

@tnorimat Thank you for the review!

**Comment by buger** [comment]
Thank you for the fix! Can't wait to try it out :)

**Comment by tnorimat** [comment]
@mposolda The PR seems good. Could you check the PR?

**Comment by buger** [comment]
We have tested it locally as well, works great!

**Comment by mposolda** [comment]
Closing as the PR was not updated for the long time. Feel free to re-open once the concerns from above are addressed (missing tests and not needed `if` ). Anyone else is also welcome to send the PR for this one.

**Pull request closed without merging**